# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/embutils/not-dietlibc.diff
# Copyright (C) 2019 - 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- embutils-0.19/mknod.c.vanilla	2019-09-17 15:30:52.292000000 +0200
+++ embutils-0.19/mknod.c	2019-09-17 15:31:29.436000000 +0200
@@ -4,6 +4,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <stdlib.h>
+#include <sys/sysmacros.h>
 #include "write12.h"
 
 char* Name="mknod: ";
--- embutils-0.19/tar.c.vanilla	2019-09-17 15:32:15.112000000 +0200
+++ embutils-0.19/tar.c	2019-09-17 15:33:24.648000000 +0200
@@ -21,6 +21,8 @@
 #ifdef __sun__
 #include <sys/mkdev.h>
 #endif
+#include <sys/sysmacros.h>
+#include <sys/types.h>
 #include "write12.h"
 
 #ifdef __MINGW32__
--- embutils-0.19/Makefile.vanilla	2019-09-17 15:33:35.044000000 +0200
+++ embutils-0.19/Makefile	2019-09-17 15:35:58.784000000 +0200
@@ -10,8 +10,8 @@
 FLAGS=
 
 PRGS=sync echo chroot mkdir rmdir mknod pwd true false clear chown chgrp \
-chmod sleep tty tar basename df tee uname arch hostname env time dirname \
-domainname id ln mv cp yes which cat rm wc ls whoami mkfifo head install \
-sosrm soscp sosmv sosln soslns md5sum sleep2 allinone uniq tr mesg du \
+chmod sleep tty tar basename df tee uname arch hostname time dirname \
+id ln mv cp yes which cat rm wc ls whoami mkfifo head install \
+sosrm soscp sosmv sosln soslns sleep2 uniq tr mesg du \
 uuencode uudecode nohup nice cmp mktemp truncate test date \
 printenv chrootuid renice \
@@ -19,6 +19,6 @@
 OS:=$(shell uname)
 ifeq ($(OS),Linux)
-PRGS+=chvt dmesg kill dd write tail mount pivot_root insmod rmmod \
+PRGS+=chvt dd write tail mount pivot_root insmod rmmod \
 lsmod touch umount
 endif
 
--- embutils-0.19/ls.c.vanilla	2019-09-17 15:33:58.704000000 +0200
+++ embutils-0.19/ls.c	2019-09-17 15:34:11.976000000 +0200
@@ -40,6 +40,7 @@
 #include <termios.h>
 #include <sys/ioctl.h>
 #include <limits.h>
+#include <sys/sysmacros.h>
 
 struct statlist {
   struct stat s;
--- embutils-0.19/kill.c.vanilla	2019-09-17 15:36:14.308000000 +0200
+++ embutils-0.19/kill.c	2019-09-17 15:36:50.416000000 +0200
@@ -90,7 +90,9 @@
   { SIGSTKFLT,	6, STR_POS( 70,  0,  0,175) },
 #endif
   { SIGCHLD,	4, STR_POS( 77, 87, 78, 78) },
+#if 0
   { SIGCLD,	3, STR_POS(156,153,153,186) },
+#endif
   { SIGCONT,	4, STR_POS( 82, 82,112,120) },
   { SIGSTOP,	4, STR_POS( 87, 72,102,108) },
   { SIGTSTP,	4, STR_POS( 92, 77,107,115) },
--- embutils-0.19/dd.c.vanilla	2019-09-17 15:37:52.120000000 +0200
+++ embutils-0.19/dd.c	2019-09-17 15:38:50.516000000 +0200
@@ -6,6 +6,10 @@
 
 #include <sys/uio.h>
 
+#ifndef PAGE_SIZE
+#define PAGE_SIZE 4096
+#endif
+
 /*
  * if=FILE	read from FILE instead of stdin
  * of=FILE	write to  FILE instead of stdout
